src/integration.test.ts

```diff
@@ -59,69 +89,89 @@ describe("Integration Test", (): void => {
     });
   });
 
-  const mockedExec = async (load: boolean, command: string): Promise<void> => {
-    const stdout = "standard output from Bash command";
-    const stderr = "error output from Bash command";
-    child_process.exec.mockImplementationOnce(
-      (_command: any, _options: any, callback: any): any => {
-        callback(null, { stdout, stderr });
-      }
+  const assertExecBashCommand = (
+    infoCallNum: number,
+    execCallNum: number,
+    command: string,
+    stdout: string
+  ): void => {
+    expect(core.info).nthCalledWith<[string]>(infoCallNum, command);
+    expect(child_process.exec).nthCalledWith<[string, ExecOptions, any]>(
+      execCallNum,
+      command,
+      EXEC_OPTIONS,
+      expect.anything()
     );
+    expect(core.info).nthCalledWith<[string]>(infoCallNum + 1, stdout);
+    expect(core.error).lastCalledWith(STDERR);
+    expect(core.setFailed).not.toHaveBeenCalled();
+  };
 
-    await (load ? docker.loadDockerImages() : docker.saveDockerImages());
-
+  const assertLoadDockerImages = (cacheHit: boolean): void => {
     expect(core.getInput).nthCalledWith<[string, InputOptions]>(1, "key", {
       required: true,
     });
-    expect(core.info).nthCalledWith<[string]>(1, command);
-    expect(child_process.exec).lastCalledWith(
-      command,
-      EXEC_OPTIONS,
-      expect.anything()
+    expect(core.setOutput).lastCalledWith(docker.CACHE_HIT, cacheHit);
+    expect(child_process.exec).toHaveBeenCalledTimes(1);
```

Comment:
Move this line after the branch so that the corresponding called withs come first.